lib/health.js

```diff
@@ -48,17 +48,35 @@ class HealthCheck {
     };
   }  
 
+  http() {
+    return (request, response) => {
+      const requestPath = request.url;
+      if (requestPath !== this.getHealthUri()) return false;
+
+      (async () => {
+        const responseContent = await this.healthResponse(this.opts);
+        const httpCode = (response.status === STATUS_ERR ) ? 503 : 200;
+        
+        response.writeHead(httpCode, { 'Content-Type': 'application/health+json; charset=utf-8' });
+        response.end(JSON.stringify(responseContent));
+      })();
+
+      return true;
+    };
+  }
+
   addCheck (componentName, metricName, checkerPromise) {
     const key = `${componentName}:${metricName}`;
-    this.checks[key] = checkerPromise;
+    this.checks[key] = {};
+    this.checks[key].executor = checkerPromise;
   }
 
   async healthResponse(opts) {
     const response = {};
     let overallStatus = STATUS_OK;
     for (const prop in this.checks) {
```

Comment:
You are performing a synchronous loop of each check, is this intentional?  My understanding of this code is that if you have 3 checks, you will execute the first check and wait for the response, exec the second check and wait for the response and then exec the 3rd check and wait for the response then finish.  I think it should execute all checks async `Promise.all` style.

Reply:
I agree with this, though it'll require some rework to ensure that none of them throw.   Right now if the executor throws, the `response` from L75 is not used.

Reply:
Great point. I will address this in the next PR, since that one is entirely about executors.

Thank you